runtime/Cpp/runtime/src/misc/IntervalSet.h

```diff
@@ -30,15 +31,17 @@ namespace misc {
   protected:
     /// The list of sorted, disjoint intervals.
     std::vector<Interval> _intervals;
-    bool _readonly;
+    std::atomic<bool> _readonly;
```

Comment:
Why do we have a mutex in the ATN for protecting the read-only state change **and** make the _readonly flag atomic? That's not necessary - one of the two can be removed. I'd prefer `atomic` to stay.